### ISSUE TITLE
Not OK: perl v5.7.0 +DEVEL7825 on powerpc-machten 4.1.4

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2892]
Migrated from [rt.perl.org#4766](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4766) (status was 'resolved')

Searchable as RT4766$

**Comment by p5pRT** [comment]
### From domo@computer.org



lib/filt\-util\.\.\.\.\.\.\.\.FAILED at test 2

The problem is that perl for MachTen is statically linked\, but does
not get built to include Filter/Util/Call\.xs&#8203;: Configure decides which
modules should be statically linked by looking for
ext/\.\.\./Something/Something\.xs \(where \.\.\./ may be absent or represent
one or more directories\) and so misses \.\.\./Util/Call\.xs\.  Hacking
Configure to be less pernickety and to look instead for Something/\*\.xs
does not fix the problem&#8203;: lib/auto/Filter/Util/Call\.a does not get
handed to ld when perl is linked; instead it gets told to search a
non\-existent library&#8203;:

cc&#8203;: lib/auto/Filter/Util/Util\.a&#8203;: No such file or directory

So\, can I request that Paul Marquess \(and/or a MakeMaker guru \-\- I've
poked at things and looked at the docs until my eyes cross without
result\) munges the contents of ext/Filter/Util to support static
linking?

Thanks\.

<details><summary>Perl Info</summary>

```

Flags:
     category=install
     severity=none

This perlbug was built using Perl v5.7.0 - Thu Nov 23 09:50:06 WET 2000
It is being executed now by  Perl v5.7.0 - Thu Nov 23 12:03:48 WET 2000.

Site configuration information for perl v5.7.0:

Configured by domo at Thu Nov 23 12:03:48 WET 2000.

Summary of my perl5 (revision 5.0 version 7 subversion 0) configuration:
   Platform:
     osname=machten, osvers=4.1.4, archname=powerpc-machten
     uname='machten computer 5 0.5 powerpc '
     config_args='-d -Dusedevel'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=undef use5005threads=undef useithreads=undef 
usemultiplicity=undef
     useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
     use64bitint=undef use64bitall=undef uselongdouble=undef
   Compiler:
     cc='cc', ccflags ='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -I/usr/local/include',
     optimize='-O2 -fomit-frame-pointer',
     cppflags='-DNOTDEF_MACHTEN -DREG_INFTY=2047 -I/usr/local/include'
     ccversion='', gccversion='2.8.1', gccosandvers='machten4'
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, 
Off_t='off_t', lseeksize=4
     alignbytes=8, usemymalloc=y, prototype=define
   Linker and Libraries:
     ld='ld', ldflags =' -Xlstack=1048576 -L/usr/local/lib'
     libpth=/usr/local/lib /usr/lib
     libs=-lndbm -lgdbm -ldb -lm -lc -lutil
     perllibs=-lndbm -lgdbm -ldb -lm -lc -lutil
     libc=/usr/lib/libc.a, so=so, useshrplib=false, libperl=libperl.a
   Dynamic Linking:
     dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
     cccdlflags='', lddlflags=''

Locally applied patches:
     DEVEL7825


@INC for perl v5.7.0:
     lib
     /usr/local/lib/perl5/5.7.0/powerpc-machten
     /usr/local/lib/perl5/5.7.0
     /usr/local/lib/perl5/site_perl/5.7.0/powerpc-machten
     /usr/local/lib/perl5/site_perl/5.7.0
     /usr/local/lib/perl5/site_perl/5.6.0/powerpc-machten
     /usr/local/lib/perl5/site_perl/5.6.0
     /usr/local/lib/perl5/site_perl
     .


Environment for perl v5.7.0:
     HOME=/home/domo
     LANG (unset)
     LANGUAGE (unset)
     LD_LIBRARY_PATH=/usr/lib
     LOGDIR (unset)
 
PATH=/sbin:/usr/sbin:/home/domo/bin:/usr/local/bin:/usr/bin:/bin:/usr/X11/bin:/usr/libexec
     PERL_BADLANG (unset)
     SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

On Thu\, Nov 23\, 2000 at 03&#8203;:45&#8203;:59PM \+0100\, Dominic Dunlop wrote&#8203;:
> This is a build failure report for perl from domo@&#8203;computer\.org\,
> generated with the help of perlbug 1\.33 running under perl v5\.7\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> lib/filt\-util\.\.\.\.\.\.\.\.FAILED at test 2
> 
> The problem is that perl for MachTen is statically linked\, but does
> not get built to include Filter/Util/Call\.xs&#8203;: Configure decides which
> modules should be statically linked by looking for
> ext/\.\.\./Something/Something\.xs \(where \.\.\./ may be absent or represent
> one or more directories\) and so misses \.\.\./Util/Call\.xs\.  Hacking
> Configure to be less pernickety and to look instead for Something/\*\.xs
> does not fix the problem&#8203;: lib/auto/Filter/Util/Call\.a does not get

I guess I just could push F&#8203;::U&#8203;::C \(somewhat unfortunately named module :\-\)
one directory level deeper\.

> handed to ld when perl is linked; instead it gets told to search a
> non\-existent library&#8203;:
> 
> cc&#8203;: lib/auto/Filter/Util/Util\.a&#8203;: No such file or directory
> 
> So\, can I request that Paul Marquess \(and/or a MakeMaker guru \-\- I've
> poked at things and looked at the docs until my eyes cross without
> result\) munges the contents of ext/Filter/Util to support static
> linking?
> 
> Thanks\.
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>      category=install
>      severity=none
> \-\-\-
> This perlbug was built using Perl v5\.7\.0 \- Thu Nov 23 09&#8203;:50&#8203;:06 WET 2000
> It is being executed now by  Perl v5\.7\.0 \- Thu Nov 23 12&#8203;:03&#8203;:48 WET 2000\.
> 
> Site configuration information for perl v5\.7\.0&#8203;:
> 
> Configured by domo at Thu Nov 23 12&#8203;:03&#8203;:48 WET 2000\.
> 
> Summary of my perl5 \(revision 5\.0 version 7 subversion 0\) configuration&#8203;:
>    Platform&#8203;:
>      osname=machten\, osvers=4\.1\.4\, archname=powerpc\-machten
>      uname='machten computer 5 0\.5 powerpc '
>      config\_args='\-d \-Dusedevel'
>      hint=recommended\, useposix=true\, d\_sigaction=define
>      usethreads=undef use5005threads=undef useithreads=undef 
> usemultiplicity=undef
>      useperlio=undef d\_sfio=undef uselargefiles=define usesocks=undef
>      use64bitint=undef use64bitall=undef uselongdouble=undef
>    Compiler&#8203;:
>      cc='cc'\, ccflags ='\-DNOTDEF\_MACHTEN \-DREG\_INFTY=2047 \-I/usr/local/include'\,
>      optimize='\-O2 \-fomit\-frame\-pointer'\,
>      cppflags='\-DNOTDEF\_MACHTEN \-DREG\_INFTY=2047 \-I/usr/local/include'
>      ccversion=''\, gccversion='2\.8\.1'\, gccosandvers='machten4'
>      intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
>      d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
>      ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, 
> Off\_t='off\_t'\, lseeksize=4
>      alignbytes=8\, usemymalloc=y\, prototype=define
>    Linker and Libraries&#8203;:
>      ld='ld'\, ldflags =' \-Xlstack=1048576 \-L/usr/local/lib'
>      libpth=/usr/local/lib /usr/lib
>      libs=\-lndbm \-lgdbm \-ldb \-lm \-lc \-lutil
>      perllibs=\-lndbm \-lgdbm \-ldb \-lm \-lc \-lutil
>      libc=/usr/lib/libc\.a\, so=so\, useshrplib=false\, libperl=libperl\.a
>    Dynamic Linking&#8203;:
>      dlsrc=dl\_none\.xs\, dlext=none\, d\_dlsymun=undef\, ccdlflags=''
>      cccdlflags=''\, lddlflags=''
> 
> Locally applied patches&#8203;:
>      DEVEL7825
> 
> \-\-\-
> @&#8203;INC for perl v5\.7\.0&#8203;:
>      lib
>      /usr/local/lib/perl5/5\.7\.0/powerpc\-machten
>      /usr/local/lib/perl5/5\.7\.0
>      /usr/local/lib/perl5/site\_perl/5\.7\.0/powerpc\-machten
>      /usr/local/lib/perl5/site\_perl/5\.7\.0
>      /usr/local/lib/perl5/site\_perl/5\.6\.0/powerpc\-machten
>      /usr/local/lib/perl5/site\_perl/5\.6\.0
>      /usr/local/lib/perl5/site\_perl
>      \.
> 
> \-\-\-
> Environment for perl v5\.7\.0&#8203;:
>      HOME=/home/domo
>      LANG \(unset\)
>      LANGUAGE \(unset\)
>      LD\_LIBRARY\_PATH=/usr/lib
>      LOGDIR \(unset\)
>  
> PATH=/sbin&#8203;:/usr/sbin&#8203;:/home/domo/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/X11/bin&#8203;:/usr/libexec
>      PERL\_BADLANG \(unset\)
>      SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

At 10&#8203;:01 AM 11/23/00 \-0600\, Jarkko Hietaniemi wrote&#8203;:
\<Snipped>
 >I guess I just could push F&#8203;::U&#8203;::C \(somewhat unfortunately named
 >module  :\-\) one directory level deeper\.

Since dos\-djgpp has the same problem\, it would help there as well\.

Maybe it would be a good idea not to abbreviate that module name quite 
like that any more\.\.\. :\-\)



---
Peter J\. Farley III \(pjfarley@&#8203;dorsai\.org OR
&nbsp;                      pjfarley@&#8203;banet\.net\)

**Comment by p5pRT** [comment]
### From @gsar

On Thu\, 23 Nov 2000 10&#8203;:01&#8203;:45 CST\, Jarkko Hietaniemi wrote&#8203;:
>On Thu\, Nov 23\, 2000 at 03&#8203;:45&#8203;:59PM \+0100\, Dominic Dunlop wrote&#8203;:
>> The problem is that perl for MachTen is statically linked\, but does
>> not get built to include Filter/Util/Call\.xs&#8203;: Configure decides which
>> modules should be statically linked by looking for
>> ext/\.\.\./Something/Something\.xs \(where \.\.\./ may be absent or represent
>> one or more directories\) and so misses \.\.\./Util/Call\.xs\.  Hacking
>> Configure to be less pernickety and to look instead for Something/\*\.xs
>> does not fix the problem&#8203;: lib/auto/Filter/Util/Call\.a does not get
>
>I guess I just could push F&#8203;::U&#8203;::C \(somewhat unfortunately named module :\-\)
>one directory level deeper\.

I'd vote for renaming it Filter&#8203;::Call instead\.  \(I don't see what the
:&#8203;:Util buys us at all\.\)


Sarathy
gsar@&#8203;ActiveState\.com

**Comment by p5pRT** [comment]
### From @jhi

On Thu\, Nov 23\, 2000 at 09&#8203;:08&#8203;:26AM \-0800\, Gurusamy Sarathy wrote&#8203;:
> On Thu\, 23 Nov 2000 10&#8203;:01&#8203;:45 CST\, Jarkko Hietaniemi wrote&#8203;:
> >On Thu\, Nov 23\, 2000 at 03&#8203;:45&#8203;:59PM \+0100\, Dominic Dunlop wrote&#8203;:
> >> The problem is that perl for MachTen is statically linked\, but does
> >> not get built to include Filter/Util/Call\.xs&#8203;: Configure decides which
> >> modules should be statically linked by looking for
> >> ext/\.\.\./Something/Something\.xs \(where \.\.\./ may be absent or represent
> >> one or more directories\) and so misses \.\.\./Util/Call\.xs\.  Hacking
> >> Configure to be less pernickety and to look instead for Something/\*\.xs
> >> does not fix the problem&#8203;: lib/auto/Filter/Util/Call\.a does not get
> >
> >I guess I just could push F&#8203;::U&#8203;::C \(somewhat unfortunately named module :\-\)
> >one directory level deeper\.
> 
> I'd vote for renaming it Filter&#8203;::Call instead\.  \(I don't see what the
> :&#8203;:Util buys us at all\.\)

At the moment\, backward compatibility with any modules that use it\.\.\.

**Comment by p5pRT** [comment]
### From @gsar

On Thu\, 23 Nov 2000 11&#8203;:16&#8203;:39 CST\, Jarkko Hietaniemi wrote&#8203;:
>On Thu\, Nov 23\, 2000 at 09&#8203;:08&#8203;:26AM \-0800\, Gurusamy Sarathy wrote&#8203;:
>> I'd vote for renaming it Filter&#8203;::Call instead\.  \(I don't see what the
>> :&#8203;:Util buys us at all\.\)
>
>At the moment\, backward compatibility with any modules that use it\.\.\.

Good point\.  We could provide a Filter/Util/Call\.pm stub for that\.

That may eventually protect us from any \(inevitable?\) incompatible
changes brought on by merging the concept of "source filter" and
"I/O discipline"\.


Sarathy
gsar@&#8203;ActiveState\.com
